### ISSUE TITLE
scrapy interop

### ISSUE THREAD

**Issue by mainrs** [hhursev/recipe-scrapers#170]
Hey!
I am currently using scrapy for its rich set of middlewares that I need to use. I was wondering how to incorporate this library into scrapy. Is there a way to pass down the `scrapy.Response` object and use that to somehow construct a parsed result? 
The only thing that should be needed are the body and the url (which are both present inside the `scrapy.Response` object).

**Comment by hhursev** [comment]
Sounds like that will be a useful addition! I'll need more details from you to be really helpful. However, if you have the content of the page you want to extract info from, you can do something along these lines:

```
# get the correct scraper for the domain you are scraping - for example AllRecipes
from recipe_scrapers.allrecipes import AllRecipes

# create a dummy file object
import io
dummy_file = io.StringIO('<html><body>Html content of the page</body><html>')

scrape_me = AllRecipes(dummy_file, test=True)
scrape_me.title()
scrape_me.ingredients()
# etc.
```

**Comment by SirWindfield** [comment]
The thing about scrapy is that is actually does download the html itself and returns the body content which can be used to query information on it.

I guess, for simple interop, a method like `scrape_me` that takes in the body response and url. The url would be used to get the right parser and the body content is used instead of making the request by itself.

**Comment by micahcochran** [comment]
I'm working on my own web crawler (not scrapy based) and having some difficulty.

The back-end interface to has seems to changed so that the above code no longer works.

With a crawler the goal is to download the webpage as few times as possible (only once is preferred).  The crawler has to download the page and parse the anchor tags in order to fine more pages to visit on the website.  The crawler would then pass the already downloaded webpage to `recipe_scrapers`.

It would be very helpful to have a public interface that bypass the `requests.get()` function either (1) by passing a url and the HTML contents (as a string or file object) to a function, or (2) by pass the `Response` object from the `requests.get()` function, which has `.text` and `.url` methods.

Having a public interface that support this functionality would be helpful.  Thank you!

**Comment by micahcochran** [comment]
Here's the v13 update for the above code:
```python
# get the correct scraper for the domain you are scraping - for example AllRecipes
from recipe_scrapers.allrecipes import AllRecipes
from recipe_scrapers.settings import settings

# create a dummy file object
import io
dummy_file = io.StringIO('<html><body>Html content of the page</body><html>')

settings.TEST_MODE = True
scrape_me = AllRecipes(dummy_file)
scrape_me.title()
scrape_me.ingredients()
# etc.
```

That back-end works okay for my web crawler.  It would be nice to have a public interface.

**Comment by jayaddison** [comment]
@mainrs @micahcochran I'm trawling old GitHub issues to close at the moment, so apologies if I haven't completely understood the description and context (I'm skim-reading).

I think that this use case is now mostly-supported by our public `scrape_html` interface (from [release 14.0.0](https://github.com/hhursev/recipe-scrapers/releases/tag/14.0.0) onwards).

Although it doesn't accept a `Response` object (neither `scrapy`, `requests` nor `urlib`), it does accept HTML content as a string (plus, optionally, the original URL) and will attempt to scrape content based on those.

**Comment by micahcochran** [comment]
@jayaddison Thank you for updating the issue.  I'll try to take a look at in when I have time, which will be a while.

**Comment by hhursev** [comment]
Will become easier with v15. Closing due to inactivity
